common/src/main/java/com/microsoft/identity/common/internal/numberMatch/NumberMatchHelper.kt

```diff
@@ -35,41 +38,62 @@ import com.microsoft.identity.common.logging.Logger
  */
 class NumberMatchHelper {
 
-    // Store number matches in a static hash map
-    // No need to persist this storage beyond the current broker process, but we need to keep them
-    // long enough for AuthApp to call the broker api to fetch the number match
+    // Store number matches in a Content Provider
     companion object {
         val TAG = NumberMatchHelper::class.java.simpleName
-        val numberMatchMap: HashMap<String, String> = HashMap()
         const val SESSION_ID_ATTRIBUTE_NAME = "sessionID"
         const val NUMBER_MATCH_ATTRIBUTE_NAME = "numberMatch"
 
         /**
-         * Method to add a key:value pair of sessionID:numberMatch to static hashmap. This hashmap will be accessed
-         * by broker api to get the number match for a particular sessionID.
+         * Method to add a key:value pair of sessionID:numberMatch to Content Provider.
          */
-        fun storeNumberMatch(sessionId: String?, numberMatch: String?) {
```

Comment:
>storeNumberMatch [](http://example.com/codeflow?start=12&length=16)

This is called broker side right, so why the overhead of content resolver here insert here?